uportal-war/src/main/java/org/apereo/portal/layout/dlm/remoting/UpdatePreferencesServlet.java

```diff
@@ -231,6 +232,65 @@ public ModelAndView removeElement(HttpServletRequest request, HttpServletRespons
     }
 
     /**
+     * Remove the first element with the provided fname from the layout.
+     *
+     * @param request
+     * @param response
+     * @param fname
+     * @return
+     * @throws IOException
+     */
+    @RequestMapping(method = RequestMethod.POST, params = "action=removeByFName")
+    public ModelAndView removeByFName(HttpServletRequest request, 
+            HttpServletResponse response,
+            @RequestParam(value = "fname", required = true) String fname)
+            throws IOException {
+
+        IUserInstance ui = userInstanceManager.getUserInstance(request);
+        IPerson per = getPerson(ui, response);
+
+        UserPreferencesManager upm = (UserPreferencesManager) ui.getPreferencesManager();
+        IUserLayoutManager ulm = upm.getUserLayoutManager();
+
+        try {
+            String elementId = ulm.getUserLayout().findNodeId(new PortletSubscribeIdResolver(fname));
+            if (elementId != null
+                    && elementId.startsWith(Constants.FRAGMENT_ID_USER_PREFIX)
+                    && ulm.getNode(elementId)
+                            instanceof org.apereo.portal.layout.node.UserLayoutFolderDescription) {
+
+                removeSubscription(per, elementId, ulm);
+
+            } else {
+                // Delete the requested element node.  This code is the same for
+                // all node types, so we can just have a generic action.
+                if (!ulm.deleteNode(elementId)) {
+                    logger.info(
+                            "Failed to remove element ID {} from layout root folder ID {}, delete node returned false",
+                            elementId,
+                            ulm.getRootFolderId());
+                    response.sendError(HttpServletResponse.SC_FORBIDDEN);
```

Comment:
SC_FORBIDDEN should probably remain here. There has already been a check to verify the node exists in the layout. Suggest adding a conditional earlier to check if the elementId is null and return SC_BAD_REQUEST in that case.